database/neo4j/neo4j.test.js

```diff
@@ -0,0 +1,19 @@
+const { startNeo4j, closeNeo4j } = require('./neo4j')
+const neo4j = require('neo4j-driver')
+
+describe('Neo4j database', () => {
+  //needed for startNeo4j
+  neo4j.driver = jest.fn().mockReturnValue({})
+
+  //needed for closeNeo4j
+  const driver = neo4j.driver()
+
+  beforeEach(startNeo4j)
+
+  driver.close = jest.fn().mockReturnValue(Promise.resolve())
+
+  test('Should resolve when closeNeo4j is called', async () => {
+    const res = await closeNeo4j()
+    expect(res).resolves
```

Comment:
I may be wrong, but I think this test code has some problem like this:

The test might pass because what you expect is just to resolve. But what you did in 6th~13th line won't affect anything to this 17th line which actually tests something. I think you tried to get the functions that `closeNeo4j` calls. But what you actually did is creating new functions/objects with the same name. You should mock them before you require `'./neo4j'`.
How about changing `expect(res).resolves` to `expect(driver.close).toHaveBeenCalled` to make sure if the test code works?

Reply:
I tried doing that before and couldn't get any tests to pass by using 
`expect(driver.close).toHaveBeenCalledTimes(1)`
<img width="562" alt="Screen Shot 2020-08-13 at 12 48 38 AM" src="https://user-images.githubusercontent.com/45890848/90108225-d0234e00-dcfe-11ea-9355-085aae82f9ee.png">

Reply:
hmm i have some ideas, i will try them out

Reply:
ah yea `jest.mock('neo4j-driver')` is what i wwas missing